cognite/client/data_classes/contextualization.py

```diff
@@ -539,6 +540,122 @@ class PersonalProtectiveEquipmentDetectionParameters(VisionResource, ThresholdPa
+            DetectJobManager._instance = DetectJobManager()
+        return DetectJobManager._instance
+
+    def __init__(self) -> None:
+        self._warning_shown = False
+        self._has_active_job = False
+
+    def set_active_job(self) -> None:
+        self._has_active_job = True
+
+    def free_active_job(self) -> None:
+        self._has_active_job = False
+
+    def has_active_job(self) -> bool:
+        return self._has_active_job
+
+    def show_warning(self) -> None:
+        if not self._warning_shown:
+            warnings.warn(
+                "DetectJobBundle.result is calling a beta endpoint which is still in development. Breaking changes can happen in between patch versions."
+            )
+            self._warning_shown = True
+
+
+class DetectJobBundle:
+    _RESOURCE_PATH = "/context/diagram/detect/"
+    _STATUS_PATH = "/context/diagram/detect/status"
+    _WAIT_TIME = 2
+
+    def __init__(self, job_ids: List[int], cognite_client: "CogniteClient" = None):
+        # Show warning
+        DetectJobManager.instance().show_warning()
+
+        self._cognite_client = cast("CogniteClient", cognite_client)
+        if not job_ids:
+            raise ValueError("You need to specify job_ids")
+        self.job_ids = job_ids
+        self._remaining_job_ids: List[int] = []
+
+        self.jobs: List[Dict[str, Any]] = []
+
+        self._result: List[Dict[str, Any]] = []
+
+    def __repr__(self) -> str:
+        return (
+            "DetectJobBundle( job_ids="
+            + str(self.job_ids)
+            + ", self.jobs="
+            + str(self.jobs)
+            + ", self._result="
+            + str(self._result)
+            + ", self._remaining_job_ids="
+            + str(self._remaining_job_ids)
+            + ")"
+        )
+
+    def _back_off(self) -> None:
+        """
+        Linear back off, in order to limit load on our API.
+        Starts at _WAIT_TIME and goes to 10 seconds.
+        """
+        time.sleep(self._WAIT_TIME)
+        if self._WAIT_TIME < 10:
+            self._WAIT_TIME += 2
+
+    def wait_for_completion(self, timeout: int = None) -> None:
+        """Waits for all jobs to complete, generally not needed to call as it is called by result.
+        Args:
+            timeout (int): Time out after this many seconds. (None means wait indefinitely)
+            interval (int): Poll status every this many seconds.
+        """
+        start = time.time()
+        self._remaining_job_ids = self.job_ids
+        while timeout is None or time.time() < start + timeout:
+            self.jobs = self._cognite_client.diagrams._post(
+                self._STATUS_PATH, json={"items": self._remaining_job_ids}
+            ).json()["items"]
+
+            # Assign the jobs that aren't finished
+            self._remaining_job_ids = [j["jobId"] for j in self.jobs if JobStatus(j["status"]).is_not_finished()]
+
+            if self._remaining_job_ids:
+                self._back_off()
```

Comment:
In the `_back_off()` function, we increase the `_WAIT_TIME` every time it is called, which is fine. However, we should not call `_back_off()` merely because we are not finished yet. 
More customary is to back off if your API is unable to respond. That is for example, because our `/status` endpoint is overloaded. You increase the backoff time to cut your API some slack and don't overload it further.

So we should add handling of bad error codes to the `/status` endpoint and call `_back_off()` only in those events.

Reply:
Please check if you agree with this: [5cc867a](https://github.com/cognitedata/cognite-sdk-python/pull/1077/commits/5cc867af10d12ee3d0029725529fb2d27130b1e9)

Reply:
Looks good!